doc: re-adds correct suggested usage for pipes

Archie-Elvie opened the pull request:
In the [current docs](https://docs.zephyrproject.org/latest/kernel/services/data_passing/pipes.html#suggested-uses), the suggest usage for pipes is the "Flushing a Pipe’s Buffer" example code. 

This PR replaces it with the text provided in the [v2.7 docs](https://docs.zephyrproject.org/2.7.0/reference/kernel/data_passing/pipes.html?highlight=pipes#suggested-uses).

mbolivar-nordic reviewed (commented):
Hi, please fix the commit message formatting.

See here for more information: https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines

peter-mitsis commented on doc/kernel/services/data_passing/pipes.rst:
```diff
@@ -203,17 +203,11 @@ data in the pipe.
+Use a pipe to send streams of data between threads.
+
+.. note::
+    A pipe can be used to transfer long streams of data if desired. However it is often preferable to send pointers to large data items to avoid copying the data.
 
-    void monitor_thread(void)
-    {
-        while (1) {
-            ...
-            /* Critical error detected. Flush the entire pipe to reset it. */
-            k_pipe_flush(&my_pipe);
-            ...
-        }
-    }
```
I think you had good catch in observing that code block is under the "Suggested uses" section. However, removing the code block is not the right thing to do. Instead it should simply be moved up above the "Suggested uses" so that it remains associated with the "Flushing a Pipe's Buffer" section.

Archie-Elvie replied:
That is true, I'll update with this change and the commit message fix

Archie-Elvie committed:
doc: fixes errors pipes wiki entry

Moves the 'Flushing a Pipe' code example to the correct location
and section and re-adds the 'Suggested Uses' text missing
compared to v2.7.

Signed-off-by: Archie Atkinson <archie.atkinson@chiaro.co.uk>

mbolivar-nordic commented:
@andyross can you please ACK as the assignee for this PR?
